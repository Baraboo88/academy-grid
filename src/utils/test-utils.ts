import { QuestLevel, QuestModel, QuestType } from './utils';
import { createApi } from '../api';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';


export const mockQuests: QuestModel[]  = [
    {
      id: 1,
      "title": "Склеп",
      "description": "Средневековое кладбище таит в себе много страшных тайн. Местные жители говорят, что в склепе похоронен граф вампир, который по ночам выходит на охоту, чтобы испить человеческой крови. Через час солнце опустится за горизонт, успеете ли вы убить вампира и выбраться из склепа?",
      "previewImg": "img/preview-sklep.jpg",
      "coverImg": "img/cover-sklep.jpg",
      type: "horror" as QuestType,
      level: "hard" as QuestLevel,
      peopleCount: [2, 5],
      duration: 120
    },
    {
      "id": 2,
      "title": "Маньяк",
      "description": "В комнате с приглушённым светом несколько человек, незнакомых друг с другом, приходят в себя. Никто не помнит, что произошло прошлым вечером. Руки и ноги связаны, но одному из вас получилось освободиться. На стене висит пугающий таймер и запущен отсчёт 60 минут. Сможете ли вы разобраться в стрессовой ситуации, помочь другим, разобраться что произошло и выбраться из комнаты?",
      "previewImg": "img/preview-maniac.jpg",
      "coverImg": "img/cover-maniac.jpg",
      "type": "horror" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [3, 6],
      "duration": 90
    },
    {
      "id": 3,
      "title": "Ритуал",
      "description": "Тяжелый воздух угнетает, в ночи вы оказываетесь запертыми в сыром помещении вместе с другими ничего не понимающими жертвами. Сквозь щель в двери вы видите, как некто в капюшоне готовит площадку как будто для проведения мистического обряда. Удастся ли вам выбраться, пока вы не станете жертвой ритуала?",
      "previewImg": "img/preview-ritual.jpg",
      "coverImg": "img/cover-ritual.jpg",
      "type": "mystic" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [3, 5],
      "duration": 120
    },
    {
      "id": 4,
      "title": "Тайны старого особняка",
      "description": "Погрузитесь в атмосферу служебных помещений закулисья, которые хранят множество тайн и загадок. Вы окажитесь в старом особняке и увидите все, что скрывают его запутанные коридоры.",
      "previewImg": "img/preview-final-frontier.jpg",
      "coverImg": "img/cover-final-frontier.jpg",
      "type": "detective" as QuestType,
      "level": "easy" as QuestLevel,
      "peopleCount": [2, 5],
      "duration": 60
    },
    {
      "id": 5,
      "title": "Хижина в лесу",
      "description": "Вы с друзьями оказались в заброшенной хижине. Какую тайну она скрывает и как из неё выбраться? На эти вопросы вам предстоит найти ответ, чтобы вернуться домой.",
      "previewImg": "img/preview-house-in-the-woods.jpg",
      "coverImg": "img/cover-house-in-the-woods.jpg",
      "type": "mystic" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [4, 7],
      "duration": 90
    },
    {
      "id": 6,
      "title": "Фатальный эксперимент",
      "description": "Вы стоите на пороге нового научного открытия, которое перевернет судьбу человечества. Но что-то идёт не так, и ядерный реактор, который работает на полную мощность, сигнализирует о скорой поломке. Удастся ли вам починить его в отведенное время и предотвратить гибель людей в этом фатальном эксперименте?",
      "previewImg": "img/preview-fatal-exp.jpg",
      "coverImg": "img/cover-fatal-exp.jpg",
      "type": "adventures" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [5, 8],
      "duration": 120
    },

  ]
;


export const getTestStore = () => {
  const initialState = {
    quests: mockQuests,
    currentQuest: mockQuests[1],
    errorMsg: ''
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
