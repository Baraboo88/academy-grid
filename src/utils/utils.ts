import { ActiveTab } from '../reducer/reducer';

export enum QuestType {
  Adventures = 'adventures',
  Horror = 'horror',
  Mystic = 'mystic',
  Detective = 'detective',
  SciFi='sci-fi'
}

export enum QuestLevel {
  Easy = 'easy',
  Medium = 'medium',
  Hard = 'hard'
}

export enum QuestDuration {
  Sixty = 60,
  Ninety = 90,
  OneTwoZero = 120
}

export interface QuestModel {
  id: number;
  title: string;
  description: string;
  previewImg: string;
  coverImg: string;
  type: QuestType;
  level: QuestLevel;
  peopleCount: number [];
  duration: QuestDuration;
}

export interface OrderModel {
  name: string;
  peopleCount: number;
  phone: string;
  isLegal: boolean
}

export interface StateModel {
  quests: QuestModel [];
  currentQuest?: QuestModel;
  isOrderSent?: boolean;
  isResponseReceived: boolean;
  errorMsg: string;
  activeTab: ActiveTab;
}


export const getCyrillicLevel = (level: QuestLevel) => {
  switch (level) {
    case QuestLevel.Easy:
      return 'простой';
    case QuestLevel.Medium:
      return 'средний';
    case QuestLevel.Hard:
      return 'сложный';
  }
  return '';
};

export const getCyrillicType = (questType: QuestType) => {
  switch (questType) {
    case QuestType.Adventures:
      return 'Приключения';
    case QuestType.Horror:
      return 'Ужасы';
    case QuestType.Mystic:
      return 'Мистика';
    case QuestType.Detective:
      return 'Детектив';
    case QuestType.SciFi:
      return 'Sci-fi';
  }
  return '';
};


export const applyFilter = (quests: QuestModel[], filter: QuestType) => quests.filter((quest) => quest.type === filter);
