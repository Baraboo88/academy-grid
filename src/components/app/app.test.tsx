import App from './app';

import * as Enzyme from 'enzyme';
import {mount} from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import {Provider} from 'react-redux';
Enzyme.configure({adapter: new EnzymeReactAdapter()});
import toJson from 'enzyme-to-json';
import { getTestStore } from '../../utils/test-utils';

it(`App successfully rendered`, () => {
  const tree = mount(<Provider store={getTestStore()}><App/></Provider>);
  expect(toJson(tree, {mode: `deep`})).toMatchSnapshot();
});
