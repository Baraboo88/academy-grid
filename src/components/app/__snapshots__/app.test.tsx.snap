// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App successfully rendered 1`] = `
Array [
  "",
  Array [
    <header
      className="sc-pVTFL cyOVn"
    >
      <div
        className="sc-gsDKAQ dEliar sc-jrQzAO iDPSeT"
      >
        <a
          className="sc-kDTinF fEliR"
        >
          <img
            alt="Логотип Escape Room"
            className="sc-egiyK"
            height="50"
            src="logo.svg"
            width="134"
          />
        </a>
        <nav
          className="sc-crHmcD"
        >
          <ul
            className="sc-bqiRlB gmJdaa"
          >
            <li
              className="sc-ksdxgE inUVAZ"
            >
              <a
                className="sc-hBUSln bjEjAu"
                href="/"
                onClick={[Function]}
              >
                Квесты
              </a>
            </li>
            <li
              className="sc-ksdxgE inUVAZ"
            >
              <a
                className="sc-hBUSln jFYCoy"
                href="/"
                onClick={[Function]}
              >
                Новичкам
              </a>
            </li>
            <li
              className="sc-ksdxgE inUVAZ"
            >
              <a
                className="sc-hBUSln jFYCoy"
                href="/"
                onClick={[Function]}
              >
                Отзывы
              </a>
            </li>
            <li
              className="sc-ksdxgE inUVAZ"
            >
              <a
                className="sc-hBUSln jFYCoy"
                href="/"
                onClick={[Function]}
              >
                Акции
              </a>
            </li>
            <li
              className="sc-ksdxgE inUVAZ"
            >
              <a
                className="sc-hBUSln jFYCoy"
                href="/contacts"
                onClick={[Function]}
              >
                Контакты
              </a>
            </li>
          </ul>
        </nav>
        <a
          className="sc-fotOHu cVznvU"
          href="tel:88003335599"
        >
          8 (800) 333-55-99
        </a>
      </div>
    </header>,
    <main
      className="sc-gsDKAQ dEliar sc-ZOtfp dpvOtb"
    >
      <div
        className="sc-fFeiMQ bKziXJ"
      >
        <h1
          className="sc-ieecCq cMcvVY"
        >
          Выберите тематику
        </h1>
        <p
          className="sc-bkkeKt iLBZUc"
        >
          квесты в Санкт-Петербурге
        </p>
      </div>
      Array [
        <ul
          className="sc-eJwWfJ iHmuzh"
        >
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj gdVQJn"
              type="button"
            >
              <svg>
                icon-all-quests.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Все квесты
              </span>
            </button>
          </li>
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj CTRNp"
              type="button"
            >
              <svg>
                icon-adventures.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Приключения
              </span>
            </button>
          </li>
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj CTRNp"
              type="button"
            >
              <svg>
                icon-horrors.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Ужасы
              </span>
            </button>
          </li>
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj CTRNp"
              type="button"
            >
              <svg>
                icon-mystic.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Мистика
              </span>
            </button>
          </li>
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj CTRNp"
              type="button"
            >
              <svg>
                icon-detective.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Детектив
              </span>
            </button>
          </li>
          <li
            className="sc-nVkyK cKFyGc"
          >
            <button
              className="sc-hiwPVj CTRNp"
              type="button"
            >
              <svg>
                icon-scifi.svg
              </svg>
              <span
                className="sc-ehCJOs cBMPAj"
              >
                Sci-fi
              </span>
            </button>
          </li>
        </ul>,
        <ul
          className="sc-lcepkR fcDmIN"
        >
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест Склеп"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-sklep.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    Склеп
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      2–5 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      сложный
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест Маньяк"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-maniac.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    Маньяк
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      3–6 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      средний
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест Ритуал"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-ritual.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    Ритуал
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      3–5 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      легкий
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест История призраков"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-old-ceil.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    История призраков
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      5–6 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      легкий
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест Тайны старого особняка"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-final-frontier.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    Тайны старого особняка
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      2–3 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      легкий
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
          <li
            className="sc-gGCDDS"
          >
            <a
              className="sc-clIzBv lnMeiB"
              href="/quest"
              onClick={[Function]}
            >
              <article
                className="sc-faUpoM hhwigY"
              >
                <img
                  alt="квест Хижина в лесу"
                  className="sc-dvQaRk eVqXdR"
                  height="232"
                  src="img/preview-house-in-the-woods.jpg"
                  width="344"
                />
                <div
                  className="sc-Galmp fSqBGf"
                >
                  <h2
                    className="sc-fWCJzd kCIWjF"
                  >
                    Хижина в лесу
                  </h2>
                  <ul
                    className="sc-TBWPX cCEZfW"
                  >
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-person.svg
                      </svg>
                      4–7 чел
                    </li>
                    <li
                      className="sc-jIkXHa dNA-DpB"
                    >
                      <svg>
                        icon-puzzle.svg
                      </svg>
                      средний
                    </li>
                  </ul>
                </div>
              </article>
            </a>
          </li>
        </ul>,
      ]
    </main>,
    <footer
      className="sc-dkPtRN cKqpWq"
    >
      <ul
        className="sc-hKwDye cIBGNF"
      >
        <li
          className="sc-eCImPb gamxB"
        >
          <a
            className="sc-jRQBWg iUJxXW"
            href="https://instagram.com"
            target="_blank"
          >
            <svg
              className="sc-gKclnd edxetw"
            >
              icon-instagram-hover.svg
            </svg>
            <span
              className="sc-furwcr visually-hidden"
            >
              Инстаграм
            </span>
          </a>
        </li>
        <li
          className="sc-eCImPb gamxB"
        >
          <a
            className="sc-jRQBWg iUJxXW"
            href="https://vk.com"
            target="_blank"
          >
            <svg
              className="sc-iCfMLu cgPhip"
            >
              icon-vk-hover.svg
            </svg>
            <span
              className="sc-furwcr visually-hidden"
            >
              Вконтакте
            </span>
          </a>
        </li>
      </ul>
    </footer>,
  ],
]
`;
